src/events/broker.py

```diff
@@ -452,14 +452,24 @@ def _init_kraft_mode(self) -> None:
             self.charm.state.cluster.update({"cluster-uuid": uuid})
             self.charm.state.peer_cluster.update({"cluster-uuid": uuid})
 
-        # Controller is tasked with populating quorum uris
+        # Controller is tasked with populating quorum uris and the `controller` user password
         if self.charm.state.runs_controller:
             quorum_uris = {"controller-quorum-uris": self.charm.state.controller_quorum_uris}
             self.charm.state.cluster.update(quorum_uris)
 
+            generated_password = self.charm.workload.generate_password()
+
             if self.charm.state.peer_cluster_orchestrator:
                 self.charm.state.peer_cluster_orchestrator.update(quorum_uris)
 
+                if not self.charm.state.peer_cluster_orchestrator.controller_password:
+                    self.charm.state.peer_cluster_orchestrator.update(
+                        {"controller-password": generated_password}
+                    )
+            elif not self.charm.state.peer_cluster.controller_password:
+                # single mode, controller & leader
+                self.charm.state.cluster.update({"controller-password": generated_password})
```

Comment:
**thought/non-blocking:** I think this Kraft stuff is complicating the `events/broker.py`.
We might be able to add a `events/controller.py`, and create some custom events that the 'broker' emits.

I'm thinking like `init-kraft` or `format-storage`. That way we can move the complexity here to a new file.
FWIW, this isn't anything related to this PR, just a thought for the future.